tests/JsonSchema/Tests/Constraints/RequiredPropertyTest.php

```diff
@@ -9,8 +9,52 @@
 
 namespace JsonSchema\Tests\Constraints;
 
+use JsonSchema\Constraints\UndefinedConstraint;
+
 class RequiredPropertyTest extends BaseTestCase
 {
+
+    public function testErrorPropertyIsPopulatedForRequiredIfMissingInInput()
+    {
+        $validator = new UndefinedConstraint();
+        $document = json_decode('{
+            "bar": 42
+        }');
+        $schema = json_decode('{
+            "type": "object",
+            "properties": {
+                "foo": {"type": "number"},
+                "bar": {"type": "number"}
+            },
+            "required": ["foo"]
+        }');
+
+        $validator->check($document, $schema);
+        $error = $validator->getErrors();
+        $this->assertEquals('foo', $error[0]['property']);
```

Comment:
If `$error` is not well formatted, i.e. there is no index 0 or 'property` key does not exists test run will stop without an useful message.

Could you try to compare the whole `$error` variable or assert 0 and property exists?